packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -351,6 +356,22 @@ class TextPainter {
     _needsLayout = false;
   }
 
+  void layoutToMaxIntrinsicWidth({ double maxWidth }) {
+    if (!_needsLayout && width == maxIntrinsicWidth && width <= maxWidth)
```

Comment:
but I still don't think this is valid.

Consider text that, at maxWidth 100, has a max intrinsic width of 90, and at maxWidth 110, has a maxIntrinsicWidth of 105.

Call layoutToMaxIntrinsicWidth(maxWidth: 100). The width and maxIntrinsicWidth become 90. Then call layoutToMaxIntrinsicWidth(maxWidth: 110). The width and maxIntrinsicWidth are equal, the width is less than 90. But you still need to lay it out again.